ui/src/timeMachine/selectors/index.ts

```diff
@@ -172,6 +191,32 @@ export const getFillColumnsSelection = (state: AppState): string[] => {
   )
 }
 
+export const getMosaicFillColumnsSelection = (state: AppState): string[] => {
+  const {table} = getVisTable(state)
+  const validFillColumns = getStringColumnsMemoized(table)
+  const preference = get(
+    getActiveTimeMachine(state),
+    'view.properties.fillColumns'
+  )
+
+  if (preference === null) {
+    for (const key of validFillColumns) {
+      if (key.startsWith('_value')) {
+        return [key]
+      }
+    }
+    if (table.columnKeys.includes('_value')) {
```

Comment:
it's _generally_ better style to put single flow control `if` statements like this ahead of looping flow control statements like `for of` above. that way, if this check fails, we exit without looping over something and potentially failing. cheaper to check one thing once than to check one thing once for every item in a loop.

Reply:
Could you explain a bit more what you mean? I'm not sure I understand.

Reply:
In this code, we're checking if there's a valid column key for _value, and if not, we're checking if _value is a numeric column. The `if` case would be true for both string and numeric columns, so we don't want to check it until we're sure "_value" is not a string[], so I'm not sure if we'll be able to switch the `if` and `for of` in this case. If you think it's worth looking into, we can try to check the type of the value column instead?